server/fleet/queries.go

```diff
@@ -17,7 +17,25 @@ type QueryPayload struct {
 
 type Query struct {
 	UpdateCreateTimestamps
-	ID          uint   `json:"id"`
+	ID uint `json:"id"`
+	// TeamID to which team this query belongs. If not set, then the query belongs to the 'Global'
+	// team. The table schema for queries includes another column related to this one:
+	// `team_id_char`, this is because the unique constraint for queries is based on both the
+	// team_id and their name, but since team_id can be null (and (NULL == NULL) != true), we need
+	// to use something else to guarantee uniqueness, hence the use of team_id_char. team_id_char
+	// will be computed as string(team_id), if team_id IS NULL then team_char_id will be ''.
```

Comment:
interesting, for MDM we used `team_id = 0`

Reply:
That would be another good approach for approach this, the main downside of that would be that we won't be able to enforce a FK constraint on `team_id`